WebDiscover: Hookup AWS RDS Flow

kimlisa opened the pull request:
part of https://github.com/gravitational/teleport/issues/22130

- Implement logic for resuming discover flow from where user left of
- Replaces the old aws rds flow with the new rds flow (connect aws integration and enroll rds databases)
- Enabled integration resource access
- Fixed a db agent label matching bug

marcoandredinis commented:
We are listing RDS DB Instances and RDS DB Clusters, so the role must include the following policy:
```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "rds:DescribeDBInstances",
                "rds:DescribeDBClusters"
            ]
            "Resource": "*"
        }
    ]
}
```

marcoandredinis commented:
When setting up the Integration we should strip out the `:443` from the Provider URL
I think it gives us a better UX
If the port is anything != then `:443` then we should add it :+1:
If we decide to keep the `:443`, then we must change this method to not strip it
https://github.com/gravitational/teleport/blob/8f80db470afb84454f9af1f2c81449884d4f5c8c/lib/web/oidcidp.go#L96-L114

kimlisa committed:
Clean up aws oidc integration instructions

kimlisa committed:
Change ResourceSpec icon type from ReactElement to string

When storing state into location URL, it doesn't allow storing
ReactElement, so I changed the icon element into string that
refers to the correct icon. Also adds rds aurora tiles to
Select Resources screen.

kimlisa committed:
Fix the expected backend aws status value for RDS list

kimlisa committed:
For RDS list, allow refreshing the table

Helpful when user makes changes to the RDS instance
(eg. tags) and needs to get the most up to date listing

kimlisa committed:
Update rds db setup access text info

kimlisa committed:
Make create database dialog more consistent btwn states

kimlisa committed:
Fix label matching

Previously we required the agent matcher labels
be an exact match of registered db labels otherwise we
prevented the user from deploying an agent (which was wrong).

Now the only requirement is that the matcher labels are all
able to match against registered db labels.

kimlisa committed:
Implement resuming discover flow from where user left of

kimlisa committed:
Enable integration access and rds flow

kimlisa committed:
Strip 443 ports from cluster uri

kimlisa committed:
Use the labels returned from polling db instaed

kimlisa committed:
Various touch ups

- Make label matching error less confusing by showing
  error upon user trying to generate command
- Make label messaging clearer
- Emit errors when failing to fetch rds dbs

kimlisa commented:
@ryanclark friendly ping

kimlisa commented:
@avatus friendly ping

avatus commented on web/packages/teleport/src/Discover/Database/DownloadScript/DownloadScript.tsx:
```diff
@@ -248,32 +234,65 @@ export const Labels = ({
+            Default wildcards label allows this database service to match any
+            database.
+            <br />
+            You can define your own labels that this database service will use
+            to identify your registered database. The labels you define must
+            match with the labels that was defined for the registered database
+            (from previous step):
```
```suggestion
            You can define your own labels that this database service will use
            to identify your registered database. The labels you define must
            match the labels that were defined for the registered database
            (from previous step):
```

avatus commented on web/packages/teleport/src/Discover/Database/DownloadScript/DownloadScript.tsx:
```diff
@@ -248,32 +234,65 @@ export const Labels = ({
+          <TextIcon>
             <Icons.Warning ml={1} color="error.main" />
-            {message}
+            The matcher labels must be able to match with the labels defined for
+            the registered database. Use wildcards to match by any labels.
```
```suggestion
            The matcher labels must be able to match with the labels defined for
            the registered database. Use wildcards to match with any labels.
```

avatus commented on web/packages/teleport/src/Discover/Database/SetupAccess/SetupAccess.tsx:
```diff
@@ -207,15 +207,7 @@ function DbEngineInstructions({
-              </Link>{' '}
-              in order to be used with Database Access for RDS. To enable, users
-              must have a <Mark>rds_iam</Mark> role:
+              Users must have a <Mark>rds_iam</Mark> role:
```
I know this said "a" rds_iam role previously, and "a" is correct if we removed the role name and just said "a role". But it reads weird and if this will always be `rds_iam` here, I think we should change to `Users must have an rds_iam role"`. However, feel free to skip this as i can see it working both ways

avatus commented on web/packages/teleport/src/Discover/useDiscover.tsx:
```diff
@@ -66,25 +66,44 @@ type CustomEventInput = {
+// and then coming back to resume the flow.)
+export type DiscoverUrlLocState = {
+  // discover contains the fields necessary to be able to resume
+  // the flow from where user left of.
```
```suggestion
  // the flow from where user left off.
```

avatus commented on web/packages/teleport/src/Discover/useDiscover.tsx:
```diff
@@ -66,25 +66,44 @@ type CustomEventInput = {
+// react routes (eg. in RDS database flow, it is required of user
+// to create a AWS OIDC integration which requires changing route
+// and then coming back to resume the flow.)
+export type DiscoverUrlLocState = {
```
I think we should change this to DiscoverUrlLocationState, just to be more clear. When we are working with it and setting it above, it's called `locationState` so makes more sense to keep it that way.

avatus reviewed (commented):
Didn't get to test locally yet but the code looks great. Just a few nits

kimlisa committed:
Address CR and update test

kimlisa committed:
Merge branch 'master' into lisa/hookup-rds-flow
